routes/verify.go

```diff
@@ -318,3 +321,94 @@ func (fes *APIServer) SubmitPhoneNumberVerificationCode(ww http.ResponseWriter,
 		}
 	}
 }
+
+type VerifyEmailRequest struct {
+	PublicKey string
+	EmailHash string
+}
+
+func (fes *APIServer) VerifyEmail(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := VerifyEmailRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("SubmitPhoneNumberVerificationCode: Problem parsing request body: %v", err))
+		return
+	}
+
+	userPublicKeyBytes, _, err := lib.Base58CheckDecode(requestData.PublicKey)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("VerifyEmail: Invalid public key: %v", err))
+		return
+	}
+
+	// Now that we have a public key, update the global state object.
+	userMetadata, err := fes.getUserMetadataFromGlobalState(lib.PkToString(userPublicKeyBytes, fes.Params))
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("VerifyEmail: Problem with getUserMetadataFromGlobalState: %v", err))
+		return
+	}
+
+	validHash := fes.verifyEmailHash(userMetadata.Email, requestData.PublicKey)
+	if requestData.EmailHash != validHash {
+		_AddBadRequestError(ww, fmt.Sprintf("VerifyEmail: Invalid hash: %s", requestData.EmailHash))
+		return
+	}
+
+	userMetadata.EmailVerified = true
+
+	err = fes.putUserMetadataInGlobalState(userMetadata)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("VerifyEmail: Failed to save user: %v", err))
+		return
+	}
```

Comment:
Do we need to send back some sort of 200 response at the end of this function? or does it automatically return a 200 if nothing is sent back?

Reply:
200, empty response is the default